source/distributed/tria.cc

```diff
@@ -3282,24 +3288,20 @@ namespace parallel
                                                      cell,
                                                      cell,
                                                      p4est_coarse_cell,
-                                                     handle,
+                                                     offset,
                                                      unpack_callback);
         }
 
       --cell_attached_data.n_attached_datas;
       if (cell_attached_data.n_attached_deserialize > 0)
-        {
-          --cell_attached_data.n_attached_deserialize;
-
-          // Remove the entry that corresponds to the handle that was
-          // passed to the current function. Double check that such an
-          // entry really existed.
-          const unsigned int n_elements_removed
-            = cell_attached_data.pack_callbacks.erase (handle);
-          (void)n_elements_removed;
-          Assert (n_elements_removed == 1,
-                  ExcInternalError());
-        }
+        --cell_attached_data.n_attached_deserialize;
+
+      // deregister corresponding pack_callback function
+      // first reset with invalid entries to preserve ambiguity of handles,
+      // then free memory when all were unpacked (see below)
+      offset = dealii::numbers::invalid_unsigned_int;
+      auto &pack_callback = std::get<1> (cell_attached_data.pack_callbacks[handle]);
```

Comment:
I see now that you're actually generating an alias variable further up with `offset` that you use for modification here. We don't particularly like them because it makes code more difficult to read -- one has to remember that two variables are actually the same.

My preference would be if you created `offset` as a `const unsigned int` further up as a non-reference, and here just put the `std::get` call on the lhs. Same for `pack_callback`.